build_tools/cmake/iree_run_module_test.cmake

```
@@ -0,0 +1,369 @@
+      list(APPEND _RULE_RUNNER_ARGS ${_EXPECTED_OUTPUT_STR})
+      list(APPEND _SRC ${_OUTPUT_FILE_SRC})
+    elseif(_OUTPUT_FILE_TYPE STREQUAL ".npy")
+      # Large npy files are not stored in the codebase. Need to download them
+      # from GCS iree-model-artifacts first and store them in the following possible
+      # paths.
+      find_file(_OUTPUT_FILE_SRC_PATH
+        NAME
+          "${_RULE_EXPECTED_OUTPUT}"
+        PATHS
+          "${CMAKE_CURRENT_SOURCE_DIR}"
+          "${CMAKE_CURRENT_BINARY_DIR}"
+          "${IREE_BENCHMARK_SUITE_DIR}"
+        NO_CACHE
+        NO_DEFAULT_PATH
+      )
+      # If the expected output npy file is not found (the large file is not
+      # loaded from GCS to `IREE_BENCHMARK_SUITE_DIR` benchmark suite test),
+      # report error.
+      if(NOT _OUTPUT_FILE_SRC_PATH)
+        message(SEND_ERROR "${_RULE_EXPECTED_OUTPUT} is not found in\n\
+          ${CMAKE_CURRENT_SOURCE_DIR}\n\
+          ${CMAKE_CURRENT_BINARY_DIR}\n\
+          ${IREE_BENCHMARK_SUITE_DIR}\n\
+          Please check if you need to download it first.")
+      else()
+        get_filename_component(_OUTPUT_FILE_SRC "${_OUTPUT_FILE_SRC_PATH}" REALPATH)
+        list(APPEND _RULE_RUNNER_ARGS "--expected_output=@${_OUTPUT_FILE_SRC}")
+        list(APPEND _SRC ${_OUTPUT_FILE_SRC})
+      endif()
+    else()  # The expected output is listed in the field.
+      list(APPEND _RULE_RUNNER_ARGS "--expected_output=\"${_RULE_EXPECTED_OUTPUT}\"")
+    endif()
+  endif(_RULE_EXPECTED_OUTPUT)
+
+  if(NOT DEFINED _RULE_DRIVER)
+    return()
+  endif()
+
+  # Dump the flags into a flag file to avoid CMake's naive handling of spaces
+  # in expected output.
+  if(_RULE_RUNNER_ARGS)
+    set(_OUTPUT_FLAGFILE "${CMAKE_CURRENT_BINARY_DIR}/${_RULE_NAME}_flagfile")
+    # Write each argument in a new line.
+    string(REPLACE ";" "\n" _OUTPUT_FLAGS "${_RULE_RUNNER_ARGS}")
+    file(WRITE "${_OUTPUT_FLAGFILE}" "${_OUTPUT_FLAGS}")
+    list(APPEND _SRC "${CMAKE_CURRENT_BINARY_DIR}/${_RULE_NAME}_flagfile")
+  endif()
+
+  # A target specifically for the test.
+  add_custom_target("${_NAME}" ALL)
+
+  iree_native_test(
+    NAME
+      "${_RULE_NAME}"
+    DRIVER
+      "${_RULE_DRIVER}"
+    SRC
+      "${_RUNNER_TARGET}"
+    ARGS
+      "--flagfile=${_OUTPUT_FLAGFILE}"
+    DATA
+      "${_SRC}"
+    LABELS
+      ${_RULE_LABELS}
+    TIMEOUT
+      ${_RULE_TIMEOUT}
+  )
+
+  # Replace dependencies passed by ::name with iree::package::name
+  iree_package_ns(_PACKAGE_NS)
+  list(TRANSFORM _RULE_DEPS REPLACE "^::" "${_PACKAGE_NS}::")
+  if(_RULE_DEPS)
+    add_dependencies(${_NAME}
+      ${_RULE_DEPS}
+    )
+  endif()
+
+  # Set expect failure cases.
+  if(_PLATFORM IN_LIST _RULE_XFAIL)
+    set(_XFAIL TRUE)
+  else()
+    set(_XFAIL FALSE)
+  endif()
+  iree_package_path(_PACKAGE_PATH)
+  set(_TEST_NAME "${_PACKAGE_PATH}/${_RULE_NAME}")
+  if(_XFAIL)
+    set_property(TEST ${_TEST_NAME} PROPERTY WILL_FAIL TRUE)
+  endif()
```

Why do you evaluate the boolean and then separately use it as a condition?

```suggestion
  iree_package_path(_PACKAGE_PATH)
  set(_TEST_NAME "${_PACKAGE_PATH}/${_RULE_NAME}")
  if(_PLATFORM IN_LIST _RULE_XFAIL)
    set_property(TEST ${_TEST_NAME} PROPERTY WILL_FAIL TRUE)
  endif()
```